### PR TITLE
*: refactor Handle in tablecodec and rowcodec

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve?

Problem Summary:

refactor for cluster index support in the future.

### What is changed and how it works?

What's Changed:
- Change the rowcodec.decoder `handleColID int64` field to `handleColIDs []int64`.
- Extract `tryDecodeHandleColumn` method for the three decoder types.
- Use Handle interface instead of int64 in rowcodec and tablecodec package.

How it Works:
pure refactor, no logic change.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- pure refactor

### CODE REVIEW

File: store/helper/helper.go [hunk 1]
```diff
@@ -309,7 +309,11 @@ func NewFrameItemFromRegionKey(key []byte) (frame *FrameItem, err error) {
 	frame.TableID, frame.IndexID, frame.IsRecord, err = tablecodec.DecodeKeyHead(key)
 	if err == nil {
 		if frame.IsRecord {
-			_, frame.RecordID, err = tablecodec.DecodeRecordKey(key)
+			var handle kv.Handle
+			_, handle, err = tablecodec.DecodeRecordKey(key)
```
Comment (cfzjywxk) [inline store/helper/helper.go:313]: How to use `DecodeRecordKey`  for this, the user expects that the `key` to be decoded into `cluster format`, but receives an `IntHandle`, and the error returned is nil or vice versa

File: util/rowDecoder/decoder.go [hunk 2]
```diff
@@ -143,6 +138,22 @@ func (rd *RowDecoder) DecodeAndEvalRowWithMap(ctx sessionctx.Context, handle int
...
 }
 
+func (rd *RowDecoder) tryDecodeFromHandle(dCol Column, handle kv.Handle) bool {
+	if handle == nil {
+		return false
+	}
+	colInfo := dCol.Col.ColumnInfo
+	if dCol.Col.IsPKHandleColumn(rd.tbl.Meta()) {
+		if mysql.HasUnsignedFlag(colInfo.Flag) {
+			rd.mutRow.SetValue(colInfo.Offset, uint64(handle.IntValue()))
```
Comment (cfzjywxk) [inline util/rowDecoder/decoder.go:148]: There is another `SetValue` outside, maybe it's to easy to get confused when to do `set`.  Or just use the decoded value as the return value, and let the caller set or put these value outside?

File: store/helper/helper.go (hunk 1 above)
Comment (coocood) [reply store/helper/helper.go:313]: This PR only do the refactor, the handle is always IntHandle.

File: util/rowDecoder/decoder.go (hunk 2 above)
Comment (coocood) [reply util/rowDecoder/decoder.go:148]: Then the code would have more indent.

The `continue` follows the pattern at line 99.

**Comment by lysu** [comment]
@leoppro maybe take care of this rowcodec API for cdc in master

**Pull request merged**